### PR TITLE
APP_ID is now optional

### LINKED ISSUE #64: APP_ID should be optional

### PR DESCRIPTION
**Pull request by Alexis-Jacob**
close #64

### CODE REVIEW

File: demo.py [hunk 1]
```diff
@@ -45,9 +45,8 @@ def demo(client=None):
     # In both ways `user_agent_prefix` parameter is optional but recommended to identify your app to the API
     # Here we actually use a mix of those two methods to illustrate:
     if client is None:
-        app_id = os.getenv('DEEPOMATIC_APP_ID')
         api_key = os.getenv('DEEPOMATIC_API_KEY')
-        client = Client(app_id, api_key, user_agent_prefix='{}-demo/{}'.format(__title__, __version__))
+        client = Client(api_key, user_agent_prefix='{}-demo/{}'.format(__title__, __version__))
```
Comment (maingoh) [inline demo.py]: You must specify `api_key=api_key`

File: deepomatic/api/http_helper.py [hunk 2]
```diff
@@ -121,9 +121,11 @@ def __init__(self, app_id=None, api_key=None, verify_ssl=None,
 
         headers = {
             'User-Agent': self.user_agent,
-            'X-APP-ID': self.app_id,
             'X-API-KEY': self.api_key,
         }
+        if self.app_id:
+            headers['X-APP-ID'] = self.app_id
```
Comment (maingoh) [inline deepomatic/api/http_helper.py]: You will need to pass your own headers into the constructor. Or create a function :
```
def default_headers(self):
    return {
            'User-Agent': self.user_agent,
            'X-APP-ID': self.app_id,
            'X-API-KEY': self.api_key,
        }
```

You can thus override it later.

File: deepomatic/api/http_helper.py [hunk 3]
```diff
@@ -121,9 +121,11 @@ def __init__(self, app_id=None, api_key=None, verify_ssl=None,
...
+
         self.session = requests.Session()
         self.session.headers.update(headers)
```
Comment (maingoh) [inline deepomatic/api/http_helper.py]: `self.session.headers.update(self.default_headers())`

**Review by maingoh — COMMENTED** [review]
Can you remove this line https://github.com/Deepomatic/deepomatic-client-python/blob/ba5877e025970eb40258a98724f9f05afec9ce7d/deepomatic/api/client.py#L41
and update the README.md (maybe explain app_id is now optional and api_key must be provided with named parameter)

**Review by maingoh — APPROVED** [review]

File: deepomatic/api/http_helper.py [hunk 4]
```diff
@@ -75,9 +75,9 @@ def __init__(self, app_id=None, api_key=None, verify_ssl=None,
             app_id = os.getenv('DEEPOMATIC_APP_ID')
         if api_key is None:
             api_key = os.getenv('DEEPOMATIC_API_KEY')
-        if app_id is None or api_key is None:
-            raise DeepomaticException("Please specify 'app_id' and 'api_key' either by passing those values to the client"
-                                      " or by defining the DEEPOMATIC_APP_ID and DEEPOMATIC_API_KEY environment variables.")
+        if api_key is None:
+            raise DeepomaticException("Please specify 'api_key' either by passing those values to the client"
```
Comment (thomas-riccardi) [inline deepomatic/api/http_helper.py]: `s/those values/it/`

File: deepomatic/api/http_helper.py [hunk 5]
```diff
@@ -75,9 +75,9 @@ def __init__(self, app_id=None, api_key=None, verify_ssl=None,
...
+                                      " or by defining the DEEPOMATIC_API_KEY environment variables.")
```
Comment (thomas-riccardi) [inline deepomatic/api/http_helper.py]: `s/variables/variable/`

File: deepomatic/api/http_helper.py (hunk 4 above)
Comment (maingoh) [reply deepomatic/api/http_helper.py]: done in https://github.com/Deepomatic/deepomatic-client-python/pull/69

File: deepomatic/api/http_helper.py (hunk 5 above)
Comment (maingoh) [reply deepomatic/api/http_helper.py]: done in done in https://github.com/Deepomatic/deepomatic-client-python/pull/69

**Pull request merged**